addons/web/static/src/js/views/calendar/calendar_renderer.js

```diff
@@ -422,15 +422,23 @@ return AbstractRenderer.extend({
                     element.find('.fc-content').after($('<div/>', {class: 'fc-bg'}));
                 }
 
-                // For month view: Show background for all-day/multidate events only
                 if (view.name === 'month' && event.record) {
                     var start = event.r_start || event.start;
                     var end = event.r_end || event.end;
                     // Detect if the event occurs in just one day
                     // note: add & remove 1 min to avoid issues with 00:00
                     var isSameDayEvent = start.clone().add(1, 'minute').isSame(end.clone().subtract(1, 'minute'), 'day');
-                    if (!event.record.allday && isSameDayEvent) {
-                        element.addClass('o_cw_nobg');
+                    if (isSameDayEvent && !self.hideTime) {
+                        if (event.showTime) {
+                            // Note: used fixed time format to display time in 12 hour format, we can
+                            // use time format from language configuration
+                            let displayTime = start.format('hh:mm A');
+                            element.find('.fc-content .fc-time').text(displayTime);
+                        }
+                        // For month view: Show background for all-day/multidate events only
+                        if (!event.record.allday) {
```

Comment:
why do you do this only if hideTime is false? Not sure it's related...

Reply:
@aab-odoo changes are Done :+1: